### PR TITLE
Rename Rust evaluator `event_property_contains` field value_type to value too

### PR DESCRIPTION
**Pull request by bnjbvr**
Following what happened in https://github.com/matrix-org/synapse/pull/15781.

### Pull Request Checklist

* [x] Pull request is based on the develop branch
* [ ] Pull request includes a [changelog file](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#changelog). The entry should:
  - Be a short description of your change which makes sense to users. "Fixed a bug that prevented receiving messages from other servers." instead of "Moved X method from `EventStore` to `EventWorkerStore`.".
  - Use markdown where necessary, mostly for `code blocks`.
  - End with either a period (.) or an exclamation mark (!).
  - Start with a capital letter.
  - Feel free to credit yourself, by adding a sentence "Contributed by @github_username." or "Contributed by [Your Name]." to the end of the entry.
* [x] Pull request includes a [sign off](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#sign-off)
* [x] [Code style](https://matrix-org.github.io/synapse/latest/code_style.html) is correct
  (run the [linters](https://matrix-org.github.io/synapse/latest/development/contributing_guide.html#run-the-linters))

### CODE REVIEW

**Comment by clokep** [comment]
Does this fix something or is it just refactoring?

File: rust/src/push/mod.rs [hunk 1]
```diff
@@ -405,7 +405,7 @@ pub struct EventPropertyIsTypeCondition {
     pub key: Cow<'static, str>,
     // During serialization, the pattern_type property gets replaced with a
     // pattern property of the correct value in synapse.push.clientformat.format_push_rules_for_user.
-    pub value_type: Cow<'static, EventMatchPatternType>,
```
Comment (clokep) [inline rust/src/push/mod.rs:408]: This doesn't look right, the the code still searches for `value_type` at https://github.com/matrix-org/synapse/pull/15781/files#diff-50b62bc5724dfc17c692c9467f9a1f92af6ecb2b2afdc41cf96939e794d12f59R68

Comment (bnjbvr) [reply rust/src/push/mod.rs:408]: I'm not familiar at all with this code, so I might be super wrong. Is this using an internal format that is live-converted to what the spec mandates? If so, then does the rust module use the internal format or the spec format?

Comment (clokep) [reply rust/src/push/mod.rs:408]: Yes it is an internal format, the code I linked to is the serialization code which converts it.

**Comment by bnjbvr** [comment]
> Does this fix something or is it just refactoring?

I thought I spotted an error in passing that could have been hard to figure, but sounds like I was wrong! Closing.

**Comment by clokep** [comment]
> > Does this fix something or is it just refactoring?
> 
> I thought I spotted an error in passing that could have been hard to figure, but sounds like I was wrong! Closing.

No worries! If you're seeing the `foo_type` over the Client-Server API definitely shout and we can look deeper into it.

**Pull request closed without merging**